packages/workflow/src/workflow.ts

```diff
@@ -545,6 +545,17 @@ export function getExternalWorkflowHandle(workflowId: string, runId?: string): E
         if (state.info === undefined) {
           throw new IllegalStateError('Uninitialized workflow');
         }
+        // TODO: deprecate this patch after "enough" time has passed
+        if (patched('__temporal_internal_connect_external_handle_cancel_to_scope')) {
+          const scope = CancellationScope.current();
+          if (scope.cancellable) {
+            scope.cancelRequested.catch(reject);
+          }
+          if (scope.consideredCancelled) {
+            return;
+          }
```

Comment:
I'd find comments helpful—perhaps these?

```suggestion
          // reject if current scope gets cancelled
          const scope = CancellationScope.current();
          if (scope.cancellable) {
            scope.cancelRequested.catch(reject);
          }
          
          // don't send `requestCancelExternalWorkflowExecution` command if current scope is already cancelled
          if (scope.consideredCancelled) {
            return;
          }
```

Reply:
I added some more informative comments, I'm not a huge fan of comments that spell out exactly what the code does, if those are required it points to bad abstractions or naming.